src/wallet/wallet.cpp

```diff
@@ -4448,10 +4448,18 @@ void CMerkleTx::SetMerkleBranch(const uint256& block_hash, int posInBlock, int b
 
 int CMerkleTx::GetDepthInMainChain(interfaces::Chain::Lock& locked_chain) const
 {
+    if (pwallet == nullptr)
+        return 0;
+
     if (hashUnset())
         return 0;
 
-    return locked_chain.getBlockDepth(hashBlock) * (nIndex == -1 ? -1 : 1);
+    if (nIndex == -1) {
+        // Tx is no more conflicted but it's not more confirmed, so return 0
```

Comment:
In commit "Use CWallet::m_last_block_processed_height in GetDepthInMainChain" (233ae92f2d51333f158810d85d13cf63a1144e6b)

This looks right but is the logic is kind of obscure. Maybe you could add an explanatory comment like "If nIndex is -1 and the block hash is set, it means the transaction is conflicted, and that m_block_height will be set to the height of the earliest conflicting block, so can return 0 is that block has been disconnected."

I think it'd also be good to replace nested `if(A) { if (B) return; }` with `if (A && B) { return; } `

Reply:
Commented and updated logic as suggested on 573ca36